official/resnet/keras/keras_cifar_main.py

```diff
@@ -177,7 +177,10 @@ def run(flags_obj):
 
   validation_data = eval_input_dataset
   if flags_obj.skip_eval:
-    tf.keras.backend.set_learning_phase(1)
+    if strategy:
+      # TODO(haoyuzhang): Understand slowdown of setting learning phase when
+      # not using distribution strategy.
+      tf.keras.backend.set_learning_phase(1)
```

Comment:
what happens if you just skip setting this altogether?

Reply:
That's a great question. I ran a set of experiments with this removed altogether last night but that ended up making benchmarks with mirrored strategy slower.

sponge/ede73a6b-979c-4a01-a609-ddfbef90694c

I will get to understand how the learning phase (and control flow ops) affect our performance.

Reply:
It is slightly unexpected that this would affect mirrored strategy, because we already hard code the learning phase internally: 
https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/python/keras/engine/training_arrays.py?l=149